### ISSUE TITLE
GRPOTrainer: Relative LLM-as-a-judge rankings

### ISSUE THREAD

**Issue by austinmw** [huggingface/trl#4707]
### Feature request

Is it currently possible to use LLM-as-a-judge with groupwise relative rankings as in OpenPipe's ART RULER?
https://art.openpipe.ai/fundamentals/ruler

If so would like to request an example, and if not would like to request this feature!

### Motivation

Space of problems the GRPOTrainer becomes useful for expands greatly if LLM-as-a-judge ranking works well, but it seems LLM-as-a-judge performs a lot better providing relative rather than absolute rankings

### Your contribution

Not sure — possibly if guided!

**Comment by qgallouedec** [comment]
I guess so, you should be able to do it by just wrapping your judge in a function that matches the expected signature


```python
def judge_reward(prompts, completions, **kwargs):
    rewards = ... # your judge logic
    return rewards

trainer = GRPOTrainer(
    ...,
    reward_funcs=judge_reward
)
```

**Comment by qgallouedec** [comment]
Just to close this one: reward functions receive the full batch of `prompts` and `completions` for the whole generation batch (not just one), so you can implement RULER-style relative/groupwise ranking directly inside your own reward function, e.g. by chunking every `num_generations` completions and ranking within each chunk.
Closing since this is achievable with the existing API without any core changes. I'll also close https://github.com/huggingface/trl/pull/6680 as I think it's sufficiently documented in this issue. Thanks!
